### ISSUE TITLE
`ExUnit.configuration()[:include]` has incomplete info

### ISSUE THREAD

**Issue by nathanl** [elixir-lang/elixir#9947]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): Elixir 1.10.2 (compiled with Erlang/OTP 21)
* Operating system: MacOS

### Current behavior

- If I run `mix test test/some_test.exs --include chromedriver` and examine `ExUnit.configuration()[:include]`, it's `[:chromedriver]`. ✅ 
- If I run `mix test test/some_test.exs:10 --include chromedriver` and examine `ExUnit.configuration()[:include]`, it's `[line: "10"]`. The `--include chromedriver` is ignored. ❌ 

### Expected behavior

If I run `mix test test/some_test.exs:10 --include chromedriver` and examine `ExUnit.configuration()[:include]`, I'd expect to see `[:chromedriver, line: "10"]`, or `[chromedriver: true, line: "10"]`. ❌

**Comment by josevalim** [comment]
Where are you accessing ExUnit.configuration? In the test_helper?

**Comment by josevalim** [comment]
Ignore my previous question, I misunderstood the problem.

It is still unclear to me if this is expected behavior or not. If we emitted `[:chromedriver, line: "10"]`, then you would run the test on line 10 PLUS all chromedriver tests and I don't think this would be the desired behaviour.

I could actually see a case for `mix test test/some_test.exs:10 --exclude chromedriver` not running any test (assuming line 10 is tagged as chromedriver), but I don't see a case for including anything beyond the current line. Thoughts?

**Comment by nathanl** [comment]
I should give some background. I was dealing with a test suite where tests tagged `chromedriver` rely on having a separate headless chrome process running, and by default, those tests are excluded.

The test in question said "if the tests weren't run with the `--include chromedriver` flag, assume chromedriver isn't running and expect a call to `MockChromedriver`". If I ran `mix test test/some_test.exs:10 --include chromedriver` it would assume that headless chrome was not running, based on the behavior above. I noticed this because the test failure was different depending on whether I specified a line number or not.

> If we emitted [:chromedriver, line: "10"], then you would run the test on line 10 PLUS all chromedriver tests and I don't think this would be the desired behaviour.

No, that definitely wouldn't be good. It seems like our test suite's usage is just misguided - it expected this data to be a list of the options passed rather than a determination, based on those options, of which tests would be run.
